Checking.rb

```diff
@@ -0,0 +1,43 @@
+module Bank
+  # this will inherit behavior from Account class with '<'
+  class CheckingAccount < Account
+
+    attr_reader :checks
+    def initialize(id, balance, open_date)
+      super
+      @checks_used = 0
+    end
+
+    def withdrawl(amnt_withdrawn) # add withdrawl functionality
+      fee_and_withdrawl = amnt_withdrawn + 100  # subtract $2 from balance (add to amnt_withdrawn)
+      if @balance - fee_and_withdrawl < 0  # Can't go negative.
+        # Outputs warning and un-modified balance
+        puts "You can't take out more $$ than you have ($#{@balance})! Don't forget, there is a $1 transaction fee."
+        return @balance
+      else
+      super(fee_and_withdrawl)
+      end
+      return @balance
+    end
+
+    def withdrawl_using_check(amount)
+      if @checks_used >= 3 # Can use 3 checks for free
+        check_fee = 200
+      else
+        check_fee = 0 # No fee for less than three checks
+        @checks_used += 1
+      end
+        if amount + check_fee <= @balance + 1000
```

Comment:
Watch your indentation here - you don't need an additional indent for the second `if` statement